invenio_pages/records/errors.py

```diff
@@ -7,28 +7,32 @@
 
 """Errors."""
 
+from flask_babelex import gettext as _
+
 
 class PageNotFoundError(Exception):
     """Page not found exception."""
 
     def __init__(self, identifier):
-        """Constructor."""
-        self.identifier = identifier
-
-    @property
-    def description(self):
-        """Exception's description."""
-        return f"The page identified by {self.identifier} cannot be found."
+        """Initialise error."""
+        super().__init__(
+            _(
+                "The page identified by {identifier} cannot be found.".format(
```

Comment:
You can use  `f` notation for formatting, since this is running in python >= 3 . Applies to other occurrences in this PR.

```suggestion
                f"The page identified by {self.identifier} cannot be found."
```

Reply:
I'm not sure about this but I think that using f templating for strings was not interacting properly with flask_babelex, but I cannot remember why.

Reply:
it is probably because f-strings are evaluated in runtime, and gettext is gathering all strings for translations before the f-string gets evaluated ? you can confrim that by checking with lazy_gettext: https://pythonhosted.org/Flask-BabelEx/#using-translations